internal/app/components/courseSelections/courseSelections.go

```diff
@@ -0,0 +1,57 @@
+package courseSelections
+
+import (
+	"github.com/gzcharleszhang/course-planner/internal/app/components/courses"
+	"github.com/gzcharleszhang/course-planner/internal/app/components/plans"
+	"github.com/rs/xid"
+)
+
+type CourseSelectionId string
+type CourseSelectionName string
+
+type CourseSelection struct {
+	Id             CourseSelectionId
+	Name           CourseSelectionName
+	TermSelections []*courses.TermSelection
+	Plans          []plans.Plan
+}
+
+func newCourseSelectionId() CourseSelectionId {
+	return CourseSelectionId(xid.New().String())
+}
+
+func NewCourseSelection(name CourseSelectionName) *CourseSelection {
```

Comment:
We prob don't need any "constructors" since the struct is public anyways, we can leave it tho and maybe it will be useful later on